File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.cpp
class Solution {
public:
    int maximumNumberOfStringPairs(vector<string>& words) {
        int ans = 0;
        unordered_map<string, int> mp;
        for(auto w: words){
            string r = w;
            reverse(r.begin(), r.end());
            if(mp[r] > 0){ ans++; mp[r]--; }
            else mp[w]++;
        }
        return ans;
    }
};

File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.java
class Solution {
    public int maximumNumberOfStringPairs(String[] words) {
        int ans = 0;
        int n = words.length;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (words[i].charAt(0) == words[j].charAt(1) && words[i].charAt(1) == words[j].charAt(0)) {
                    ans++;
                }
            }
        }
        return ans;
    }
}